### PR TITLE
DIGI_8_GetAllAddsOns

### PR DESCRIPTION
**Pull request by bommavikas**
1)Created AddONController with Index Method to geta all avaialable AddOns
2)Created View for Index
3)Created AddOnLibrary
4)Created AddOnModel
5)Added Navigation in _HeaderMenu View
6)Added EndPoint
7)Modified LibraryUtility

### CODE REVIEW

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+    {
+        AddOnLibrary addOnLibrary = null;
+
+        /// Author: Vikas Bomma
+        /// Date: 7/14/2020
+        /// Description: Get all AddOns from database
+        /// Ref. StoryId: DIGI 8
+        /// Reviewer: 
+        /// Version: 1.0
+
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs:23]: Remove space

File: DigiOrderEnterprise/DigiOrderEnterprise/Views/AddOn/Index.cshtml [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+<div style="text-align:center;background-color:red">
+    <span style="color:white">@ViewBag.ErrorMessage</span>
+</div>
+<div style="text-align:center;background-color:green">
+    <span style="color:white">@TempData["smessage"]</span>
+</div>
+<div>
+    <div class="entry value-plus active">&nbsp;</div>
+</div>
+@{ if (Model != null)
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Views/AddOn/Index.cshtml:15]: Check model count >0

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+            try
+            {
+                addOnLibrary = new AddOnLibrary();
+                Task<HttpResponseMessage> response = addOnLibrary.getAllAddOnS();
+                string data = response.Result.Content.ReadAsStringAsync().Result;
+                modelList = JsonConvert.DeserializeObject<List<AddOnModel>>(data);
+            }
+            catch (Exception ex)
+            {
+                ViewBag.ErrorMessage = "Unable To Get The Data";
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs:36]: change message "Something went wrong.Contact administrator.."

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs (hunk 1 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs]: Removed Space

File: DigiOrderEnterprise/DigiOrderEnterprise/Views/AddOn/Index.cshtml (hunk 2 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Views/AddOn/Index.cshtml]: Added Model Count in View

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs (hunk 3 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Controllers/AddOnController.cs]: Changed Message

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/TaxPageController.cs [hunk 4]
```diff
@@ -1,37 +1,15 @@
 ﻿using DigiOrderEnterpriseLibrary;
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Controllers/TaxPageController.cs:1]: @bommavikas  revert this taxpagecontroller

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 5]
```diff
@@ -45,30 +45,27 @@
     <WarningLevel>4</WarningLevel>
   </PropertyGroup>
   <ItemGroup>
-    <Reference Include="DigiOrderEnterpriseLibrary, Version=1.0.0.0, Culture=neutral, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
+    <Reference Include="DigiOrderEnterpriseLibrary">
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:48]: Check this .csproj agian for refference check devbranch

File: DigiOrderEnterprise/DigiOrderEnterprise/Views/Shared/Partial/_HeaderMenu.cshtml [hunk 6]
```diff
@@ -64,7 +64,6 @@
                             <div class="row">
                                 <div class="multi-gd-img">
                                     <ul class="multi-column-dropdown">
-                                        <li><a href="~/TaxPage">TaxPage</a></li>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Views/Shared/Partial/_HeaderMenu.cshtml:67]: Keep reference of taxpage as it is

File: DigiOrderEnterprise/DigiOrderEnterprise/Web.config [hunk 7]
```diff
@@ -12,7 +12,6 @@
 	  <add key="digiapiurl" value="http://localhost/DigiOrderApi/api/" />
 	  <add key="digiapiuserid" value="ENTERPRISE" />
 	  <add key="digiapipassword" value="B75A5773-E01A-422C-9958-6194D8E9D0F8"/>
-    <add key="digiapigrant_type" value="password"/>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Web.config:15]: Dont remove grandtype

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj [hunk 8]
```diff
@@ -58,4 +59,4 @@
     <None Include="packages.config" />
   </ItemGroup>
   <Import Project="$(MSBuildToolsPath)\Microsoft.CSharp.targets" />
-</Project>
\ No newline at end of file
+</Project>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:62]: Aff EOF

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/EndPoint.cs [hunk 9]
```diff
@@ -1,11 +1,10 @@
-﻿
-namespace DigiOrderEnterpriseLibrary
+﻿namespace DigiOrderEnterpriseLibrary
 {
     class EndPoint
     {
         public const string ALL_CATEGORIES = "Category";
         public const string NEW_CATEGORY = "Token";
         public const string NEW_TAXPAGE = "Taxpage/create";
-        public const string ALL_TAXPAGE = "Taxpage/All";
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/EndPoint.cs:9]: Add all taxpage end point

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs [hunk 10]
```diff
@@ -15,6 +15,7 @@ class LibraryUtility
     {
 
         HttpResponseMessage response = null;
+
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs:18]: Remice space

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs [hunk 11]
```diff
@@ -23,23 +24,22 @@ class LibraryUtility
...
             try
             {
                 using (var client = new HttpClient())
                 {
-                    var val = getTokenDetails();
-                    client.BaseAddress = new Uri(apiUrl + endPoint);
-                    client.DefaultRequestHeaders.Clear();
-                    client.DefaultRequestHeaders.Add("Authorization", "bearer" + " " + val.ElementAt(0).Value);
-                    response = client.GetAsync(apiUrl + endPoint).Result;
+                    client.BaseAddress = new Uri(apiUrl);
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs:35]: Please check the functionality of getapisync

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/TaxPageLibrary.cs [hunk 12]
```diff
@@ -8,7 +8,7 @@ namespace DigiOrderEnterpriseLibrary
 {
     public class TaxPageLibrary
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/TaxPageLibrary.cs:9]: Revert taxpage library

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj (hunk 8 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:62]: EOF Available

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/EndPoint.cs (hunk 9 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/EndPoint.cs:9]: Added TaxPage EndPoint

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 5 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Checked And Resolved

File: DigiOrderEnterprise/DigiOrderEnterprise/Views/Shared/Partial/_HeaderMenu.cshtml (hunk 6 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Views/Shared/Partial/_HeaderMenu.cshtml]: Kept Taxpage Reference

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs (hunk 10 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs]: Removed Space

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/EndPoint.cs (hunk 9 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/EndPoint.cs:9]: Added TaxPage EndPoint

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs (hunk 11 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs]: Checked

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj (hunk 8 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/DigiOrderEnterpriseLibrary.csproj:62]: EOF Available

File: DigiOrderEnterprise/DigiOrderEnterprise/Web.config (hunk 7 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Web.config:15]: Resolved

File: DigiOrderEnterprise/DigiOrderEnterprise/Web.config [hunk 13]
```diff
@@ -9,7 +9,7 @@
       <add key="webpages:Enabled" value="false" />
       <add key="ClientValidationEnabled" value="true" />
       <add key="UnobtrusiveJavaScriptEnabled" value="true" />
-	  <add key="digiapiurl" value="http://localhost/DigiOrderApi/api/" />
+	  <add key="digiapiurl" value="http://localhost:52893/api/" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/Web.config:12]: add /Digiorderapi not with  portnumber

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 14]
```diff
@@ -67,7 +67,7 @@
     <Reference Include="System.Data.DataSetExtensions" />
     <Reference Include="System.Xml.Linq" />
     <Reference Include="System.Web" />
-    <Reference Include="System.Web.Extensions" />
+	<Reference Include="System.Web.Extensions" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:70]: Remove two line space before reference

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 15]
```diff
@@ -105,6 +105,7 @@
     <Reference Include="Newtonsoft.Json">
       <HintPath>..\packages\Newtonsoft.Json.12.0.2\lib\net45\Newtonsoft.Json.dll</HintPath>
     </Reference>
+    <Reference Include="System.Xml.Linq" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:108]: Remove reference

File: DigiOrderEnterprise/DigiOrderEnterprise/Controllers/TaxPageController.cs (hunk 4 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Controllers/TaxPageController.cs]: @anebhargavkrishna Tax Page Reverted

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/TaxPageLibrary.cs [hunk 16]
```diff
@@ -27,8 +27,8 @@ public Task<HttpResponseMessage> CreateTaxPage(string Title, string Description)
             }
             return httpResponse;
         }
-		
-		 public Task<HttpResponseMessage> getAllTaxPages()
+
+        public Task<HttpResponseMessage> getAllTaxPages()
         {
             try
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/TaxPageLibrary.cs:33]: Revert taxpage libraray

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/TaxPageLibrary.cs (hunk 12 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/TaxPageLibrary.cs]: Reverted Tax Page Library

File: DigiOrderEnterprise/DigiOrderEnterpriseModel/DigiOrderEnterpriseModel.csproj [hunk 17]
```diff
@@ -43,9 +43,14 @@
...
   <ItemGroup>
+    <Compile Include="AddOnModel.cs" />
     <Compile Include="CategoryModel.cs" />
     <Compile Include="Properties\AssemblyInfo.cs" />
     <Compile Include="TaxPageModel.cs" />
   </ItemGroup>
+  <ItemGroup>
+    <Content Include="bin\Debug\DigiOrderEnterpriseModel.dll" />
+    <Content Include="bin\Debug\DigiOrderEnterpriseModel.pdb" />
+  </ItemGroup>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseModel/DigiOrderEnterpriseModel.csproj:54]: Remove item group

File: DigiOrderEnterprise/DigiOrderEnterprise/Web.config (hunk 13 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/Web.config]: Added DigiOrderaapi and Removed Port Number

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 14 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Removed Spaces Before Reference

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs [hunk 18]
```diff
@@ -13,7 +13,6 @@ namespace DigiOrderEnterpriseLibrary
 {
     class LibraryUtility
     {
-
         HttpResponseMessage response = null;
         string apiUrl = Convert.ToString(ConfigurationManager.AppSettings["digiapiurl"]);
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs:17]: Change as per devbranch

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 15 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Removed Reference

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 19]
```diff
@@ -50,7 +50,7 @@
       <HintPath>..\DigiOrderEnterpriseLibrary\bin\Debug\DigiOrderEnterpriseLibrary.dll</HintPath>
     </Reference>
     <Reference Include="DigiOrderEnterpriseModel">
-      <HintPath>..\DigiOrderEnterpriseLibrary\bin\Debug\DigiOrderEnterpriseModel.dll</HintPath>
+      <HintPath>..\..\..\..\DIGI10-Create\digiorder_enterprise\DigiOrderEnterprise\DigiOrderEnterpriseModel\bin\Debug\DigiOrderEnterpriseModel.dll</HintPath>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:53]: Add removed hintpath

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 20]
```diff
@@ -51,6 +51,7 @@
     </Reference>
     <Reference Include="DigiOrderEnterpriseModel">
       <HintPath>..\DigiOrderEnterpriseLibrary\bin\Debug\DigiOrderEnterpriseModel.dll</HintPath>
+      <HintPath>..\..\..\..\DIGI10-Create\digiorder_enterprise\DigiOrderEnterprise\DigiOrderEnterpriseModel\bin\Debug\DigiOrderEnterpriseModel.dll</HintPath>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:54]: Remove line 54

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 21]
```diff
@@ -235,7 +237,9 @@
     <Content Include="asset\fonts\glyphicons-halflings-regular.ttf" />
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff" />
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff2" />
-    <Content Include="Web.config" />
+    <Content Include="Web.config">
+      <SubType>Designer</SubType>
+    </Content>
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:242]: Remove line 240 241,242 and add this  <Content Include="Web.config" />

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs [hunk 22]
```diff
@@ -23,7 +23,7 @@ class LibraryUtility
         /// <param name="endPoint"></param>
         /// <returns></returns>
         public async Task<HttpResponseMessage> GetAPISync(string endPoint)
-        {           
+        {
             try
             {
                 using (var client = new HttpClient())
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs:29]: Revert library utility do changes as per dev branch

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/TaxPageLibrary.cs (hunk 16 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/TaxPageLibrary.cs]: Reverted Tax Page Library

File: DigiOrderEnterprise/DigiOrderEnterpriseModel/DigiOrderEnterpriseModel.csproj (hunk 17 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseModel/DigiOrderEnterpriseModel.csproj]: Removed Iten Group

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs (hunk 18 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs]: Changed LibraryUtility As Per Dev Branch

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 20 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Removed Line 54

File: DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs (hunk 22 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterpriseLibrary/LibraryUtility.cs]: Reverted Library Utitity

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj (hunk 21 above)
Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Removed Line 240,241,242

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 23]
```diff
@@ -235,7 +236,6 @@
     <Content Include="asset\fonts\glyphicons-halflings-regular.ttf" />
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff" />
     <Content Include="asset\fonts\glyphicons-halflings-regular.woff2" />
-    <Content Include="Web.config" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:238]: add this deleted line

Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:238]: Added Deleted Line

File: DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj [hunk 24]
```diff
@@ -235,7 +236,7 @@
...
+	<Content Include="Web.config" />
```
Comment (anebhargavkrishna) [inline DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj:239]: check allignment

Comment (bommavikas) [reply DigiOrderEnterprise/DigiOrderEnterprise/DigiOrderEnterprise.csproj]: Checked And Resolved

**Pull request merged**